src/answers-umd.js

```diff
@@ -177,9 +177,10 @@ class Answers {
     }
 
     if (globalStorage.getState(StorageKeys.REFERRER_PAGE_URL) === null) {
-      const referrer = urlWithoutQueryParamsAndHash(document.referrer);
-      persistentStorage.set(StorageKeys.REFERRER_PAGE_URL, referrer, true);
-      globalStorage.set(StorageKeys.REFERRER_PAGE_URL, referrer);
+      globalStorage.set(
```

Comment:
Could you explain again why we need to set referrer page url and api context every time a search is run? Don't really understand why we don't want these on a redirectUrl searchbar. Even after a redirect, shouldn't referrer page url be correctly updated to the new document.referrer?  And wouldn't api context have to be manually set, so even if the searchbar had a redirectUrl wouldn't we want to carry it over?

Reply:
If we are on a client page like https://www.yext.com/ then we would automatically change the parent URL from `https://www.yext.com/` to `https://www.yext.com/?referrerPageUrl=` on ANSWERS.init. This is not the desired behavior from the PM because we should aim to not modify a client's domain (especially if it is their homepage).

Also API context can be set with `ANSWERS.setContext` on any page with Answers. Thus someone with a redirectUrl searchbar can set context by using `ANSWERS.setContext`

Reply:
I see makes sense to me!

Reply:
document.referrer is something that does not ever change on the page, so does it need to be put into globalStorage? It's also something that is recomputed on init, regardless of what the previous value in the url is, so is it something we need to put into persistentStorage? If we don't store it in either global or persistentStorage, we can remove it from the setQueryParams part of core.verticalSearch()/search() and simplify things a good deal, and we also don't have to pass it through to any static methods, and instead have those static methods call a static getter.

Reply:
`document.referrer` sadly will change as we navigate within an experience. For example if you land on the universal search, the referrerPageUrl will be `''` but if you navigate to FAQs, `document.referrer` for that page will be universal now instead of the beginning `''`. We would like to propagate the first `document.referrer` through clicks in the experience (esp. for iframes).

Reply:
in that case isnt the faq page a separate page so answers.init gets called again?

Reply:
ok sooo nevermind I didn't understand that we wanted referrer page url to be the original referrer to the experience, sgtm!